connexion/decorators/uri_parsing.py

```diff
@@ -188,7 +188,7 @@ def _make_deep_object(k, v):
         root_key = k.split("[", 1)[0]
         if k == root_key:
             return (k, v, False)
-        key_path = re.findall(r'\[(\w+)\]', k)
+        key_path = re.findall(r'\[(.[^\[\]]*)\]', k)
```

Comment:
Why do you keep a first dot (".") in there (any character)?

Reply:
So, if it was "1+ word-letters-only" before, I'd suggest to follow the same rule as "1+ of anything that is not `[]`".

 ```suggestion
        key_path = re.findall(r'\[([^\[\]]+)\]', k)
```

Nested `[]` can be sacrificed for now — it is quite non-trivial to implement opening-closing bracket matching in regexp without an extra parser.